libsolidity/analysis/TypeChecker.cpp

```diff
@@ -737,6 +737,46 @@ void TypeChecker::endVisit(ExpressionStatement const& _statement)
 			typeError(_statement.expression().location(), "Invalid integer constant.");
 }
 
+bool TypeChecker::visit(Conditional const& _conditional)
+{
+	expectType(_conditional.condition(), BoolType());
+
+	_conditional.trueExpression().accept(*this);
+	_conditional.falseExpression().accept(*this);
+
+	TypePointer const& trueType = type(_conditional.trueExpression());
```

Comment:
Please convert trueType to its mobile type (and do not store it as a reference in that case).